### ISSUE TITLE
Zsh completion: Options and flags are not completed when they come after ordered arguments

### ISSUE THREAD

**Issue by AndydeCleyre** [astral-sh/uv#3249]
```console
$ uv pip compile requirements.in --  # mash TAB key here
```

The command accepts options in this position, but that is not reflected by the current Zsh completion.

```console
$ uv --version
uv 0.1.37
$ uname
Linux
```

**Comment by zanieb** [comment]
`--` is typically used to indicate you are passing options through to another tool. Is there a reason you're using it like this?

**Comment by AndydeCleyre** [comment]
Sorry, I wasn't clear -- I'm not using dash, dash, space, but instead: dash, dash, tab

**Comment by zanieb** [comment]
Oh! Thanks for clarifying haha interesting. We don't write our own completion tooling, not sure if we can do anything here. I wonder if this reproduces with a trivial clap example or if there's an issue upstream?

**Comment by charliermarsh** [comment]
Dumb question but just confirming that you went through the steps to generate / install the Zsh completions?

**Comment by AndydeCleyre** [comment]
@charliermarsh 

Yes, of course, thanks. The completion in general works, but is more sensitive to order than the actual command.

**Comment by AndydeCleyre** [comment]
I can't tell if this is or is not describing the same issue: https://github.com/clap-rs/clap/issues/5244

**Comment by AndydeCleyre** [comment]
OK, found these:

- https://github.com/clap-rs/clap/discussions/4648
- https://github.com/clap-rs/clap/issues/3166
